pinot-core/src/main/java/com/linkedin/pinot/core/realtime/converter/stats/RealtimeNoDictionaryColStatistics.java

```diff
@@ -29,120 +29,23 @@
+    _operatorName = dataSource.getOperatorName();
     Block block = dataSource.nextBlock();
     _numDocIds = block.getMetadata().getEndDocId() + 1;
     _blockValSet = block.getBlockValueSet();
   }
 
-  private void gatherStatistics() {
-    final int rows[] = new int[_numDocIds];
-    for (int i = 0; i < _numDocIds; i++) {
-      rows[i] = i;
-    }
-    switch (_blockValSet.getValueType()) {
-      case INT:
-        computeIntMinMax(rows);
-        break;
-      case LONG:
-        computeLongMinMax(rows);
-        break;
-      case FLOAT:
-        computeFloatMinMax(rows);
-        break;
-      case DOUBLE:
-        computeDoubleMinMax(rows);
-        break;
-      default:
-        throw new UnsupportedOperationException();
-    }
-  }
-
-  private void computeIntMinMax(int[] rows) {
-    int values[] = new int[_numDocIds];
-    _blockValSet.getIntValues(rows, 0, _numDocIds, values, 0);
-    int min = Integer.MAX_VALUE;
-    int max = Integer.MIN_VALUE;
-    for (int i = 0; i < _numDocIds; i++) {
-      if (values[i] < min) {
-        min = values[i];
-      }
-      if (values[i] > max) {
-        max = values[i];
-      }
-    }
-    _minValue = Integer.valueOf(min);
-    _maxValue = Integer.valueOf(max);
-  }
-
-  private void computeLongMinMax(int[] rows) {
-    long values[] = new long[_numDocIds];
-    _blockValSet.getLongValues(rows, 0, _numDocIds, values, 0);
-    long min = Long.MAX_VALUE;
-    long max = Long.MIN_VALUE;
-    for (int i = 0; i < _numDocIds; i++) {
-      if (values[i] < min) {
-        min = values[i];
-      }
-      if (values[i] > max) {
-        max = values[i];
-      }
-    }
-    _minValue = Long.valueOf(min);
-    _maxValue = Long.valueOf(max);
-  }
-
-  private void computeFloatMinMax(int[] rows) {
-    float values[] = new float[_numDocIds];
-    _blockValSet.getFloatValues(rows, 0, _numDocIds, values, 0);
-    float min = Float.MAX_VALUE;
-    float max = Float.MIN_VALUE;
-    for (int i = 0; i < _numDocIds; i++) {
-      if (values[i] < min) {
-        min = values[i];
-      }
-      if (values[i] > max) {
-        max = values[i];
-      }
-    }
-    _minValue = Float.valueOf(min);
-    _maxValue = Float.valueOf(max);
-  }
-
-  private void computeDoubleMinMax(int[] rows) {
-    double values[] = new double[_numDocIds];
-    _blockValSet.getDoubleValues(rows, 0, _numDocIds, values, 0);
-    double min = Double.MAX_VALUE;
-    double max = Double.MIN_VALUE;
-    for (int i = 0; i < _numDocIds; i++) {
-      if (values[i] < min) {
-        min = values[i];
-      }
-      if (values[i] > max) {
-        max = values[i];
-      }
-    }
-    _minValue = Double.valueOf(min);
-    _maxValue = Double.valueOf(max);
-  }
-
   @Override
   public Object getMinValue() {
-    if (_minValue == null) {
-      gatherStatistics();
-    }
-    return _minValue;
+    throw new RuntimeException("Cannot get min value for no dictionary column " + _operatorName);
```

Comment:
Whats the impact of this change? Does the calling code appropriately bracket this in try/catch? This shouldn't result in query time exceptions.

Reply:
Looks like this is used only in SegmentColumnarIndexCreator and does not get used in the query path. As the description notes SegmentColumnarIndexCreator calls min/max only for time-column in writeMetadata().

Reply:
Even in segment creation, this call never happens (with my change). Prior to my change, the call used to happen but the returned value never used.